app/services/status_message_creation_service.rb

```diff
@@ -19,9 +19,18 @@ def create(params)
 
   def build_status_message(params)
     public = params[:public] || false
+    params[:status_message][:text] = filter_mentions params
```

Comment:
Since this is a rather internal method and it gets the entire hash, I think it's okay for it to modify it too, the current line looks a bit weird to me, "filter a hash and get a string back". So just `filter_mentions params` and change `:text` inside.